adapters/repos/db/index.go

```diff
@@ -864,6 +866,7 @@ func (i *Index) asyncReplicationEnabled() bool {
 	i.replicationConfigLock.RLock()
 	defer i.replicationConfigLock.RUnlock()
 
+	// TODO should we update this to use the new shardHasMultipleReplicas?
```

Comment:
@jeroiraz should we change this to some kind of "shard specific" setting which uses `shardHasMultipleReplicas` instead of `i.Config.ReplicationFactor`? In general, I think it would be beneficial to make async replication configured at the tenant/shard level rather than only at the collection level as it is now

Reply:
same with updateReplicationConfig?